api/views/event.py

```diff
@@ -28,3 +29,27 @@ def get_event_types(self, request, pk=None):
         Returns a dictionary of the defined event types
         """
         return Response(EVENT_TYPE_CHOICES)
+
+    @action(detail=False, methods=['post'], url_path="duplicate-event")
+    def duplicate_event(self, request):
+        """
+        Duplicates an event as well as the questions within the event.
+        """
+        event = Event.objects.filter(id=request.data.get("event", None)).first()
+        course = CanvasCourse.objects.filter(id=request.data.get("course", None)).first()
+        old_event_id = event.id
+        event.id = None
```

Comment:
Let's add a function to the event called `clone` and here you just call `event.clone()`.
Similarly you can have `question.clone()` and use it in the event clone function.